webrtc/src/rtp_transceiver/rtp_receiver/mod.rs

```diff
@@ -198,7 +198,7 @@ impl RTPReceiverInternal {
 
         let tracks = self.tracks.read().await;
         for t in &*tracks {
-            if t.track.rid() == rid {
+            if t.track.rid().map_or(false, |r| r == rid) {
```

Comment:
I think when comparing to an option `option = Some(value)` might be easier to read than `map_or`

Reply:
IIRC, you can't just do a Some(value) because the types don't line up, track.rid() is Option<String> (I toyed with having it be something else, I don't know what the rust convention is and it seemed to match the other types across the codebase) and rid is a &str, so I'd have to make it owned in order to do the comparison which seems even more of a stretch than doing a map_or, which is fundamentally equivalent to a fromMaybe in a function language like haskell/purescript.

If the left hand type can be changed then this would go away. If you have a suggestion for this then please do tell (I did start with &Option<String> to avoid the clone but that just pushed the clone to the consumers which was :vomiting_face: )

Reply:
You can do `track.rid().as_deref() == Some(rid)` but I guess that's not much better than this. By using `as_deref` the [Deref coercion](http://doc.rust-lang.org/1.64.0/std/ops/trait.Deref.html#more-on-deref-coercion) mechanism kicks in, resulting in a left hand side type of `Option<&str>`

Reply:
That's a neat little feature - would there be benefit in changing the type of track.rid()? It's mostly used in checks just like this anyway

Reply:
That's a good point actually, it should be `Option<&str>` to avoid the allocation each time it's called. The caller can still `clone` it if they want

Reply:
Yeah okay, ```Option<String>``` on the struct and ```Option<&str>``` on the trait is much nicer (and analogous to how you'd do a String anyway), I note that there are plenty of inconsistencies in how strings are passed around in the codebase in general and I've left most of them alone in this change so there are a few unnecessary String::from's here, but on the whole it's much nicer